src/eduid_common/session/namespaces.py

```diff
@@ -56,10 +57,29 @@ class MfaAction(SessionNSBase):
     authn_instant: Optional[str] = None
     authn_context: Optional[str] = None
 
+
+@dataclass()
+class TimestampedNS(SessionNSBase):
+    ts: Optional[datetime] = None
```

Comment:
ok agreed

Reply:
No sorry, this would not work, since namespaces are created empty on access, see for example [here](https://github.com/SUNET/eduid-common/blob/master/src/eduid_common/session/eduid_session.py#L94), and are generally in the session irrespective of whether they have data or not.

We would need a different, lazy mechanism to namespace the session, which of course is doable, but I would prefer doing in a PR of its own - I suspect it can have unexpected consecuences.

Reply:
Namespaces are not initialized before the property on the session is called, I would call that lazy but perhaps they could be even lazier :)

But I see what you mean, it is probably good that you can initialize an "empty" signup ns to set timestamp the first time.